src/components/ReservationContainer/ReservationContainer.component.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import { withRouter } from 'react-router-dom';
+
+import ReservationWarning from '../ReservationWarning/ReservationWarning.component';
+
+import { Reservation } from './ReservationContainer.styles';
+
+const ReservationContainer = ({ match }) => {
+	const [emailInput, setEmailInput] = useState('');
+
+	const emailInputChange = (e) => {
+		setEmailInput(e.target.value);
+	};
+
+	const { purchase_type } = match.params;
```

Comment:
```suggestion
const ReservationContainer = () => {
	const [emailInput, setEmailInput] = useState('');
	const { purchase_type } = useMatch();

	const emailInputChange = (e) => {
		setEmailInput(e.target.value);
	};
```
이렇게 하면 props로 넘겨줄 필요도 없고 withRouter HOC를 사용하지 않아도 되겠죠?

Reply:
혹시 react-router 라이브러리에 있는 함수인가요?